[laney0808] iP

laney0808 opened the pull request:
### Rick Chatbot

Rick helps you to keep track of to-dos, events and deadlines.

- add _to do items_
- add deadlines and events with date and time
- mark items as done or unmark items as undone

Rick helps you achieve **seamless** workflow.

### To use Rick ✨

1. Download [Rick Latest Version](https://github.com/laney0808/ip/releases/tag/A-Jar)
2. Wait for the initialization
3. Add your tasks by inputing command "todo [your todo item]"
4. Let Rick do the rest!
5. ~Try out erratic commands to have a taste of Rick's error messages~

- [ ] I agree to Terms and Conditions in using Rick

> "Rick helps me to remember all the things that I do not bother to." - Amily, Rick's precious client

In case you need the `main` function, it is here:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

laney0808 committed:
rename, greeting and exit

laney0808 committed:
no message

laney0808 committed:
add, list

laney0808 committed:
mark

laney0808 committed:
todo, events, deadlines

laney0808 committed:
TextUiTesting

laney0808 committed:
Handle errors

laney0808 committed:
delete

alfaloo commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,25 @@
+    }
+    public String print(){
+        return "[D]" + this.status + this.name + " (by: " + ddl + ")";
+    };
```
why does this method have a semicolon after the curly bracket

alfaloo replied:
do you think its a cleaner solution to add an overriding toString method?

alfaloo commented on src/main/java/Rick.java:
```diff
@@ -0,0 +1,134 @@
+        System.out.println(divider + arg + divider);
+    }
+
+    public static void exit(){
```
do you think it would be better to keep curly bracket placement consistent (other methods have a white space after the round bracket and before the curly bracket)

alfaloo commented on src/main/java/RickException.java:
```diff
@@ -0,0 +1,10 @@
+public class RickException extends Exception{
```
do you think you should keep consistency with spacing? the other exceptions have a white space before the curly bracket

alfaloo commented on src/main/java/ToDo.java:
```diff
@@ -0,0 +1,20 @@
+    public void unmark() {
+        this.status = "[ ]";
+    }
+}
```
do you think it would be better implementing an overriding toString method to display the tasks?

alfaloo reviewed (commented):
Overall very well done. Just some nitpicks. Keep up with the good work.

laney0808 committed:
change print to toString, add more specific error handling, clean format and spacing in output

laney0808 committed:
correct ui test based on new adjustments, improve error handling to avoid repeated error message

laney0808 committed:
add save function

laney0808 committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Rick.java
Resolve conflict by using branch-Level-7

laney0808 committed:
resolve conflict

laney0808 committed:
Implement LocalDateTime

laney0808 committed:
add new classes and implementation

laney0808 committed:
add more OOP, delete unused methods

laney0808 committed:
add packages

laney0808 committed:
Merge branch 'add-gradle-support'

laney0808 committed:
add gradle, add Junit

laney0808 committed:
add Junit tests

laney0808 committed:
change all attributes to private

laney0808 committed:
correct exception message

laney0808 committed:
add conditional to time argument for Deadline and Event

laney0808 committed:
change JUnit test based on latest changes

laney0808 committed:
change build.gradle setting to match the current project

laney0808 committed:
change build.gradle based on requirements of shadowJar

laney0808 committed:
remove 'public' for attributes in Item interface

laney0808 committed:
add javadoc for all public methods

laney0808 committed:
modify based on coding standard

laney0808 committed:
add and test find method

laney0808 committed:
Merge branch 'a-coding-standard', resolve conflicts

# Conflicts:
#	src/main/java/rick/Rick.java
#	src/main/java/rick/tasks/Item.java

laney0808 committed:
Merge branch 'branch-level-9'

laney0808 committed:
add checkstyle

laney0808 committed:
correct style

laney0808 committed:
reconstruct the project; use JavaFX and FXML

laney0808 committed:
add command classes, parser and executer; implement new execution for bye, list, delete, (un)mark, and find

laney0808 committed:
implement executer

laney0808 committed:
check style

laney0808 committed:
debug GUI issues, work in progress

laney0808 committed:
fix implementation problems; add empty GUI without handling

laney0808 committed:
add simple GUI

laney0808 committed:
DialogBox: text is in system font

The system font is ugly.

Change text label font

The font of text in the user's and rick's messages is set to Baloo Da 2 Regular for aesthetic effect.

laney0808 committed:
Parser and Command classes's methods may encounter unexpected input.

Causes program to terminate unexpectedly.

Add assertion statements at necessary lines.

Print out error in GUI when error occurs.

laney0808 committed:
Some methods do not have documentation.

This could cause confusion to person reading the code.

Add documentation for all methods.

Some method's documentation is of incorrect format.

Consistency is lost.

Change document methods to standardized format.

laney0808 committed:
Incorrect indentation of Java doc comment incurs checkstyle warning.

Add indentation to Java doc comments.

laney0808 committed:
Merge pull request #2 from laney0808/branch-A-Assertions

Parser and Command classes's methods may encounter unexpected input.

laney0808 committed:
Merge branch 'master' into branch-A-CodeQuality

laney0808 committed:
Merge pull request #3 from laney0808/branch-A-CodeQuality

Branch code quality

laney0808 committed:
Add chechClash method within Event class to check if two events clash.

laney0808 committed:
Add better GUI.

laney0808 committed:
Add personality to Rick chatbot so it sounds like Rick himself.

Replace all of Rick's responses and error messages to AI-generated Rick-like speech.

laney0808 committed:
UI: Add new icon

The app has no icon and can be difficult to looking for

Let's:
* Add a new icon to the main window

UI: Change style of send button

The text on the button says 'Button', making its purpose unclear

Let's:
* Set the text on button as "Send"
* Change button's color to themed color
* Let button's color change on hover and click

laney0808 committed:
RickException Class: Change error message

Some edge cases are not considered when parsing deadline and event command,
and error messages are misleading to the user

Let's:
* Change the error message to be less misleading

laney0808 committed:
Exit: Add automatic exit when ByeCommand is received

Window closes only when manually closed, while user expects window to close on ByeCommand

Let's:
* Exit the application automatically when ByeCommand is input

Executer Class: Change name

Name of this class is wrong (should be 'Executor') and is very similar to Java library class Executors

Let's:
* Change the name to CommandExecutor

laney0808 committed:
Exit: Exit after goodbye message is shown

Window closes upon sending 'bye', but user expects to see their own dialog box and Rick's response before exiting

Let's:
* Schedule Platform.exit() 1 second after

laney0808 committed:
Exit: Change exit method

When bye command is sent, the window closes but the application does not terminate

Let's:
* Change exit statement from Platform.exit() to System.exit(0) to make sure the application terminates

laney0808 committed:
Add User Guide

Adjust size of dialog box

The dialog box does not expand as expected when find command is run

Let's
* Standardize the output format of find command with other commands to avoid this

laney0808 committed:
Update README.md

laney0808 committed:
Update README.md

laney0808 committed:
Update README.md

laney0808 committed:
Update README.md

The User Guide has no installation guide which may confuse the users who are not familiar with Java and Jar files

Let's:
* Add a brief installation guide
